contracts/verifiers/ZKPVerifier.sol

```diff
@@ -4,16 +4,22 @@ pragma solidity 0.8.16;
 import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 import {GenesisUtils} from "../lib/GenesisUtils.sol";
-import {PoseidonFacade} from "../lib/Poseidon.sol";
 import {ICircuitValidator} from "../interfaces/ICircuitValidator.sol";
 import {IZKPVerifier} from "../interfaces/IZKPVerifier.sol";
+import {ArrayUtils} from "../lib/ArrayUtils.sol";
 
 contract ZKPVerifier is IZKPVerifier, Ownable {
+    /**
+     * @dev Max return array length for request queries
+     */
+    uint256 public constant REQUEST_QUERIES_RETURN_LIMIT = 1000;
+
     // msg.sender-> ( requestID -> is proof given )
     mapping(address => mapping(uint64 => bool)) public proofs;
 
-    mapping(uint64 => ICircuitValidator.CircuitQuery) public requestQueries;
-    mapping(uint64 => ICircuitValidator) public requestValidators;
+    mapping(uint64 => IZKPVerifier.ZKPRequest) public requestQueries;
+
+    IZKPVerifier.ZKPRequest[] public requestQueriesArr;
 
     uint64[] internal _supportedRequests;
```

Comment:
we can store only one field 
 IZKPVerifier.ZKPRequest[] public requestQueriesArr;

    uint64[] internal _supportedRequests;
   
   e.g. rename requestQueriesArr to _supportedRequests
   also make a method for getZKPRequestsCount(), so you can use pagination.. or introduce a public variable.